File: examples/java/kora-java-petclinic/src/main/java/io/koraframework/example/petclinic/repository/VisitRepository.java
package io.koraframework.example.petclinic.repository;

import io.koraframework.database.common.annotation.Id;
import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcRepository;
import io.koraframework.example.petclinic.model.dao.Visit;
import java.util.List;

@Repository
public interface VisitRepository extends JdbcRepository {

    @Query("SELECT %{return#selects} FROM %{return#table} WHERE pet_id = :petId ORDER BY visit_date, id")
    List<Visit> findByPetId(long petId);

    @Id
    @Query("INSERT INTO %{entity#inserts -= id}")
    long insert(Visit entity);
}

File: examples/kotlin/kora-kotlin-petclinic/src/main/kotlin/io/koraframework/kotlin/example/petclinic/repository/VisitRepository.kt
package io.koraframework.kotlin.example.petclinic.repository

import io.koraframework.database.common.annotation.Id
import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository
import io.koraframework.kotlin.example.petclinic.model.Visit

@Repository
interface VisitRepository : JdbcRepository {
    @Query("SELECT %{return#selects} FROM %{return#table} WHERE pet_id = :petId ORDER BY visit_date, id")
    fun findByPetId(petId: Long): List<Visit>

    @Id
    @Query("INSERT INTO %{entity#inserts -= id}")
    fun insert(entity: Visit): Long
}
